### ISSUE TITLE
Automatic Forge dependency installation

### ISSUE THREAD

**Issue by apjanke** [apjanke/octave-packajoozle#14]
```
>> pkj install -forge image symbolic optim io
pkj: installing: image 2.10.0, symbolic 2.7.1, optim 1.5.3, io 2.4.12
Installed image 2.10.0 from Octave Forge to user pkg dir
For information about changes from previous versions of the image package, run 'news image'.
Installed symbolic 2.7.1 from Octave Forge to user pkg dir
For information about changes from previous versions of the symbolic package, run 'news symbolic'.
error: pkj: unsatisified dependencies: struct >= 1.0.12, statistics >= 1.4.0
error: called from
    require_deps_installed_from_desc at line 275 column 9
    install_pkg_from_file_impl at line 200 column 7
    install_forge_pkg_single at line 150 column 11
    install_forge_pkgs at line 120 column 9
    pkj>install_forge_packages at line 365 column 3
    pkj at line 327 column 9
```

Wah, wah.

Automatically resolve and install dependencies when installing with `-forge`. Consider doing so for file installations, too.

**Comment by apjanke** [comment]
I added a dependency resolver in https://github.com/apjanke/octave-packajoozle/commit/ec32ed7647231b7f31fcf35a509e75d6399fdbbd. Now I need to hook up `pkj install` and `pkj load` to it, and hook up its `verbose` setting to the `pkj -verbose` option

**Comment by apjanke** [comment]
Done in https://github.com/apjanke/octave-packajoozle/commit/345f0f82efff1df814b041db2786cdf5735d2e1b.

```
>> addpath('~/local/repos/octave-packajoozle/inst')
>> pkj list
pkj: no packages installed
>> pkj install -forge financial
OctaveForgeClient.get_package_description_meta: cache miss  for financial 0.5.3; downloading...
DependencyResolver: working financial 0.5.3 ([financial 0.5.3])
  deps: octave >= 4.4.0; io >= 2.4.11; statistics >= 1.4.0
DependencyResolver:   considering dep: octave >= 4.4.0
DependencyResolver:     ignoring pseudopackage dependency octave >= 4.4.0
DependencyResolver:   considering dep: io >= 2.4.11
DependencyResolver:     satisfied by package found in source: io 2.4.12; adding dep
OctaveForgeClient.get_package_description_meta: cache miss  for io 2.4.12; downloading...
DependencyResolver: working io 2.4.12 ([financial 0.5.3 -> io 2.4.12])
  deps: octave >= 3.8.0
DependencyResolver:   considering dep: octave >= 3.8.0
DependencyResolver:     ignoring pseudopackage dependency octave >= 3.8.0
DependencyResolver:   considering dep: statistics >= 1.4.0
DependencyResolver:     satisfied by package found in source: statistics 1.4.0; adding dep
OctaveForgeClient.get_package_description_meta: cache miss  for statistics 1.4.0; downloading...
DependencyResolver: working statistics 1.4.0 ([financial 0.5.3 -> statistics 1.4.0])
  deps: octave >= 4.0.0; io >= 1.0.18
DependencyResolver:   considering dep: octave >= 4.0.0
DependencyResolver:     ignoring pseudopackage dependency octave >= 4.0.0
DependencyResolver:   considering dep: io >= 1.0.18
DependencyResolver:     already satisfied by already-resolved pkg order (io 2.4.12)
pkj: picked up dependencies: io 2.4.12, statistics 1.4.0, financial 0.5.3
pkj: installing: io 2.4.12, statistics 1.4.0, financial 0.5.3
warning: pkj: failed creating lookfor cache for io: '__init_io__' undefined near line 2 column 1
Installed io 2.4.12 from Octave Forge to user pkg dir
For information about changes from previous versions of the io package, run 'news io'.
Installed statistics 1.4.0 from Octave Forge to user pkg dir
For information about changes from previous versions of the statistics package, run 'news statistics'.
Installed financial 0.5.3 from Octave Forge to user pkg dir
For information about changes from previous versions of the financial package, run 'news financial'.
>> pkj list
PackageName | CurrentVersion | InstallationDir
------------+----------------+-------------------------------------------------------------------------------
financial   | 0.5.3          | /Users/janke/Library/Application Support/Octave.app/4.4.1/pkg/financial-0.5.3
io *        | 2.4.12         | /Users/janke/Library/Application Support/Octave.app/4.4.1/pkg/io-2.4.12
statistics  | 1.4.0          | /Users/janke/Library/Application Support/Octave.app/4.4.1/pkg/statistics-1.4.0
>>
```
